pilot/pkg/security/authz/matcher/header_test.go

```diff
@@ -45,11 +45,11 @@ func TestHeaderMatcher(t *testing.T) {
 		{
 			Name: "suffix match",
 			K:    ":path",
-			V:    "*/productpage*",
+			V:    "*/productpage",
```

Comment:
the old unit tests treat it as a `suffix` match, this PR treat it as a `contains` match(see L81).